### PR TITLE
Add `classes` to the `Flowers102` dataset

### PR DESCRIPTION
**Pull request by ageron**
Many datasets provide class names, why not Flowers102?
This PR adds the class names to the `Flowers102` class.

### CODE REVIEW

**Review by NicolasHug — COMMENTED** [review]
Thansk for the PR @ageron , this looks good. Just to make sure these are correct, can you share how you created the `classes` list?

**Comment by ageron** [comment]
Hi Nicolas,
The official page lists all the categories, but only shows their counts: https://www.robots.ox.ac.uk/~vgg/data/flowers/102/categories.html

So I looked for other sources and found several:
* https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/datasets/oxford_flowers102/oxford_flowers102_dataset_builder.py
* https://github.com/KlrShaK/Oxford_flowers102-using-Tensorflow/blob/master/categories_list.txt
* https://gist.github.com/JosephKJ/94c7728ed1a8e0cd87fe6a029769cde1#file-oxford-102_flower_dataset_labels-txt

They all agree. I checked that the label counts match, and they all do (adding the training set, validation set, and test set). However, some counts have duplicates:

```
40 {'prince of wales feathers', 'moon orchid', 'bolero deep blue', 'fire lily', 'pink primrose', 'canterbury bells', 'mexican aster'}
41 {'toad lily', 'grape hyacinth', 'siam tulip', 'corn poppy', 'globe-flower'}
42 {'red ginger', 'spring crocus'}
45 {'globe thistle', 'garden phlox', 'tiger lily'}
46 {'ball moss', 'monkshood', 'love in the mist'}
48 {'spear thistle', 'blackberry lily'}
49 {'oxeye daisy', 'yellow iris', 'blanket flower', 'balloon flower', 'king protea'}
52 {'silverbush', 'carnation'}
54 {'black-eyed susan', 'bearded iris', 'windflower'}
56 {'sweet pea', 'giant white arum lily', 'great masterwort'}
58 {'trumpet creeper', 'tree mallow'}
59 {'pincushion flower', 'daffodil'}
61 {'sunflower', 'osteospermum'}
63 {'desert-rose', 'magnolia', 'bromelia'}
66 {'stemless gentian', 'mallow', 'bee balm'}
67 {'marigold', 'orange dahlia', 'lenten rose', 'gaura'}
71 {'pelargonium', 'buttercup'}
78 {'artichoke', 'gazania'}
82 {'mexican petunia', 'peruvian lily', 'canna lily'}
85 {'sweet william', 'bird of paradise', 'wild pansy', 'purple coneflower'}
87 {'snapdragon', "colt's foot"}
91 {'fritillary', 'camellia'}
93 {'poinsettia', 'primula'}
109 {'pink-yellow dahlia?', 'bishop of llandaff'}
```

So it's possible that some of these are inverted. That said, I spot checked a few and they looked like the flowers on the Wikipedia, so I'm pretty confident about these labels.

**Comment by ageron** [comment]
Hi there, does this PR need any tweak or can it be merged?

**Comment by NicolasHug** [comment]
Thanks @ageron , nothing more to do on your side. I'll merge it eventually, when I have more time to dedicate to torchvision.
